evalml/automl/engine/engine_base.py

```diff
@@ -91,7 +91,7 @@ def submit_scoring_job(self, automl_config, pipeline, X, y, objectives):
         """Submit job for pipeline scoring."""
 
 
-def train_pipeline(pipeline, X, y, optimize_thresholds, objective):
+def train_pipeline(pipeline, X, y, optimize_thresholds, objective, X_schema=None, y_schema=None):
```

Comment:
Does this mean this is true for any pickling/cloudpickling scenario? Will this affect users who try to save AutoMLSearch (which has a reference to X/y_train)?

Reply:
Also, I guess this function is more for automl but what happens if the user passes in X, y with schemas already? From the code, it looks like the schema is overwritten by what's set by `X_schema`, `y_schema`?

Reply:
@angela97lin Great questions! In the case of pickling AutoMLSearch - The `X_train` and `y_train` will not  have ww init when you load from pickle but since `X_schema` and `y_schema` are saved in `self.automl_config` that information will not be lost. I think the only modification I need to make is for the engines to use the schema in `self.automl_config` for the training job. The evaluation job we're set.

Regarding your second question, these functions aren't meant to be called by users. We can make them private but I don't think we need to do that in this PR?

Reply:
> pickle/cloudpickle do not pickle the schema

This is because we don't include the training data in the picked pipeline we send to the worker, correct?

Reply:
We send the pipeline and data (X and y) to the workers. Under the hood, dask is doing something along the lines of `pickle.dump` or `cloudpickle.dump` to serialize the data before sending to the worker. This essentially nukes the woodwork accessor. Woodwork has a way to serialize ww dataframes but it's going to take some work to make sure dask knows how to properly serialize dataframes with woodwork. We filed https://github.com/alteryx/evalml/issues/2252 to track that. It seems that dask has a way to extend their serializers so that's what I hope we can do in #2252 !

Reply:
Understood. That's confusing. Thanks for explaining!